Command/CustomerChangePasswordCommand.php

```diff
@@ -37,26 +37,23 @@ class CustomerChangePasswordCommand extends Command
      * @var InputInterface
      */
     private $input;
+    
+    /**
+     * @var AppState
+     */
+    private $state;
 
     public function __construct(
         CustomerFactory $customerFactory,
         StoreManagerInterface $storeManager,
         CustomerResource $resource,
-        AppState $appState
+        AppState $state
     ) {
         parent::__construct();
         $this->customerFactory = $customerFactory;
         $this->customerResource = $resource;
         $this->storeManager = $storeManager;
-        $this->guardAgainstSessionInitializationBug($appState);
```

Comment:
I am absolutely with you that this should not be called in the constructor. That was a bad choice.
Moving the mutation to the `execute` method is a good idea.
However, I would like to keep the `try/catch` in place for backward compatibility as it was required by older Magento versions. I don't remember the specifics, but I do remember I added it for a reason.
So maybe keep `guardAgainstSessionInitializationBug` and just move the call to `execute?

Reply:
On execute try catch is worthless unless you use other session based logic. Perhaps you had issue in more complex scripts and just included it for safety.